src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,11 +1,37 @@
-export const TodoFilter = () => (
+import React from 'react';
+import { FilterBySelect } from '../../types/FilterBySelect';
+
+type Props = {
+  query: string;
+  filterBySelect: string;
+  onSetQuery: (value: string) => void;
+  onSetFilterBySelect: (value: FilterBySelect) => void;
+};
+
+export const TodoFilter: React.FC<Props> = React.memo(({
+  query,
+  filterBySelect,
+  onSetQuery,
+  onSetFilterBySelect,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
+        <select
+          data-cy="statusSelect"
+          value={filterBySelect}
+          onChange={event => {
+            onSetFilterBySelect(event.target.value as FilterBySelect);
+          }}
+        >
+          {Object.values(FilterBySelect).map(option => (
+            <option
+              value={option}
+              key={option}
+            >
+              {option[0].toUpperCase() + option.slice(1)}
+            </option>
+          ))}
```

Comment:
Smart, but hard to maintain and can cause a lot of issues in real project

Reply:
This strange solution necessary only for tests. Because they expected value in option in lower case